src/garage/torch/distributions/tanh_normal.py

```diff
@@ -0,0 +1,154 @@
+"""A Gaussian distribution with tanh transformation."""
+from collections import namedtuple
+
+import torch
+from torch.distributions import Distribution, MultivariateNormal
+
+
+class TanhNormal(Distribution):
+    # pylint: disable=abstract-method
+    """A gaussian distribution that has a tanh transformation applied to it.
+
+    Algorithms like SAC and Pearl use this transformed distribution.
+    It can be thought of as a distribution of X where
+        Y ~ N(mean, cov)
+        X ~ tanh(Y)
+
+    Args:
+        mean(torch.tensor): The mean of this distribution.
+        cov(torch.tensor): The covariance of this distribution.
+        epsilon(float): Coefficient of stability used when calculating
+                        the log likehilhood of a sample on a distribution.
+
+    """
+
+    def __init__(self, mean, cov, epsilon=1e-6):
+        self._normal = MultivariateNormal(mean, cov)
+        self._epsilon = epsilon
+        super().__init__()
+
+    def log_prob(self, value, pre_tanh_value=None):
+        # pylint: disable=arguments-differ
+        """The log likelihood of a sample on the this Tanh Distribution.
+
+        Args:
+            value(torch.tensor): The sample whose loglikelihood is being
+                                 computed.
+            pre_tanh_value(torch.tensor): The value prior to having the tanh
+                                          function applied to it but after it
+                                          has been sampled from the normal
+                                          distribution.
+
+        Notes:
+            - pre_tanh_value should usually not be None, but as the option
```

Comment:
if you don't have any usages where this is the case, please remove the option. it makes the API simpler and the test coverage easier.

Reply:
also, notes don't need bullets if there's only one of them